src/modules/flow/http-server/http-server.c

```diff
@@ -255,40 +248,84 @@ common_response_cb(void *data, struct sol_http_request *request)
         }
     }
 
-    response.response_code = SOL_HTTP_STATUS_INTERNAL_SERVER_ERROR;
-
     if (send_json) {
-        r = sol_buffer_append_printf(&response.content, "{\"%s\":", mdata->path);
-        SOL_INT_CHECK_GOTO(r, < 0, end);
+        r = sol_buffer_append_printf(&response->content,
+            "{\"%s\":", mdata->path);
+        SOL_INT_CHECK_GOTO(r, < 0, err_exit);
     }
 
-    r = type->response_cb(mdata, &response.content, send_json);
-    SOL_INT_CHECK_GOTO(r, < 0, end);
+    r = type->response_cb(mdata, &response->content, send_json);
+    SOL_INT_CHECK_GOTO(r, < 0, err_exit);
 
     if (send_json) {
-        r = sol_buffer_append_char(&response.content, '}');
-        SOL_INT_CHECK_GOTO(r, < 0, end);
+        r = sol_buffer_append_char(&response->content, '}');
+        SOL_INT_CHECK_GOTO(r, < 0, err_exit);
+    }
+
+    err_r = sol_http_param_add(&response->param, SOL_HTTP_REQUEST_PARAM_HEADER(
```

Comment:
Ugh, the code in this function even tries to return ``-EINVAL``, it's crying to have the return type changed.